### PR TITLE
feat(auth): Username/password flow

### PR DESCRIPTION
**Pull request by dnys1**
Adds support for the `USER_PASSWORD_AUTH` flow

### CODE REVIEW

File: packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart [hunk 1]
```diff
@@ -182,47 +179,30 @@ class SignInStateMachine extends StateMachine<SignInEvent, SignInState> {
...
+      case AuthFlowType.userSrpAuth:
+        expectPassword();
+        return initiateSrpAuth(event);
+      case AuthFlowType.customAuth:
+        return initiateCustomAuth(event);
+      case AuthFlowType.userPasswordAuth:
+        return initiateUserPasswordAuth(event, expectPassword());
+      case AuthFlowType.refreshToken:
+      case AuthFlowType.refreshTokenAuth:
+      case AuthFlowType.adminNoSrpAuth:
```
Comment (haverchuck) [inline packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart:200]: Why do we have the admin flow types here? These require developer credentials and I don't think would ever be in a client-side app.

**Review by haverchuck — COMMENTED** [review]
Left one question; otherwise LGTM

File: packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart (hunk 1 above)
Comment (dnys1) [reply packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart:200]: It's part of the SDK's enum

Comment (dnys1) [reply packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart:200]: I think it's just the diff formatting. The next line after the break is `throw StateError('Unsupported auth flow')`: https://github.com/aws-amplify/amplify-flutter/pull/1767/files/6add824531df9a3d7fd1336bec88acb25f5e2182#diff-9d1c23e771518da5d10ac1f888bd38c5f865868332b126a16ab6d19fd3ef028fR205

**Review by Jordan-Nelson — APPROVED** [review]

**Pull request merged**